### PR TITLE
[2.13.x] DDF-3941 Changed the user agent whitelist to a user agent blacklist in the CSRF filer

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
Changed the user agent whitelist to a user agent blacklist

#### Who is reviewing it? 
@garrettfreibott @brjeter 

#### Ask 2 committers to review/merge the PR and tag them here.
@rzwiefel
@stustison

#### How should this be tested?
- Verify that endpoints under /services that are not whitelisted cannot be accessed from a browser (eg. https://localhost:8993/services/csw?service=CSW&version=2.0.2&request=GetCapabilities)
- Verify that the endpoint can be accessed from postman or curl.
- verify that basic DDF functionalities still work (ingest, query, download, action providers)

#### What are the relevant tickets?
[DDF-3941](https://codice.atlassian.net/browse/DDF-3941)

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/filter/security-filter-csrf/src/main/java/org/codice/ddf/security/filter/csrf/CsrfFilter.java [hunk 1]
```diff
@@ -290,17 +276,14 @@ private void respondForbidden(HttpServletResponse httpResponse, String msg) {
...
-      this.userAgentWhitelist.add(userAgentPattern);
-    }
+  private boolean isNotBrowser(String userAgentHeader) {
+    return userAgentHeader == null
+        || !(userAgentHeader.contains("Mozilla")
+            || userAgentHeader.contains("Safari")
+            || userAgentHeader.contains("OPR")
+            || userAgentHeader.contains("MSIE")
+            || userAgentHeader.contains("Edge")
+            || userAgentHeader.contains("Chrome"));
```
Comment (brjeter) [inline platform/security/filter/security-filter-csrf/src/main/java/org/codice/ddf/security/filter/csrf/CsrfFilter.java]: ❓ Is there a reason you didn't just do the negated condition (here and in the if statement above) and name the method `isBrowser`? It's a little less readable with the double negation.

Comment (blen-desta) [reply platform/security/filter/security-filter-csrf/src/main/java/org/codice/ddf/security/filter/csrf/CsrfFilter.java]: I wanted to keep it similar to the ECP code

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
rebased
build now

**Comment by paouelle** [comment]
Should there be some documentation about this?

**Comment by blen-desta** [comment]
build now

**Pull request merged**